### ISSUE TITLE
Error on to_svg with embed_font=True in Spyder-IDE

### ISSUE THREAD

**Issue by ChristofKaufmann** [amueller/word_cloud#530]
#### Description
When calling `wc.to_svg(embed_font=True)` from Spyder-IDE, there occurs a ValueError. Without Spyder it works as expected.

#### Steps/Code to Reproduce

1. Install current master branch of wordcloud (release does not include the `to_svg` method yet).
1. Open Spyder.
1. Copy the following test code (into file or on console inside Spyder):
   ```python
   from wordcloud import WordCloud
   
   wc = WordCloud().generate('test')
   with open('word-cloud.svg', 'w') as out:
       out.write(wc.to_svg(embed_font=True))
   ```

#### Expected Results
SVG file generated.

#### Actual Results
Instead the following stacktrace appears:
```python
Traceback (most recent call last):

  File "/home/myuser/dev/contribute/wordcloud-fix/test/word-cloud.py", line 5, in <module>
    out.write(wc.to_svg(embed_font=True))

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/wordcloud/wordcloud.py", line 866, in to_svg
    woff.importXML(buffer)

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/ttLib/ttFont.py", line 335, in importXML
    reader.read()

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/misc/xmlReader.py", line 48, in read
    self._parseFile(self.file)

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/misc/xmlReader.py", line 73, in _parseFile
    parser.Parse(chunk, 0)

  File "/tmp/build/80754af9/python_1565725737370/work/Modules/pyexpat.c", line 471, in EndElement

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/misc/xmlReader.py", line 155, in _endElementHandler
    self.currentTable.fromXML(name, attrs, content, self.ttFont)

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/ttLib/tables/_h_e_a_d.py", line 113, in fromXML
    value = timestampFromString(value)

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/site-packages/fontTools/misc/timeTools.py", line 47, in timestampFromString
    return calendar.timegm(time.strptime(value)) - epoch_diff

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/_strptime.py", line 571, in _strptime_time
    tt = _strptime(data_string, format)[0]

  File "/home/myuser/.local/opt/miniconda3/lib/python3.7/_strptime.py", line 359, in _strptime
    (data_string, format))

ValueError: time data 'Mon Jan  8 12:28:04 2007' does not match format '%a %b %d %H:%M:%S %Y'
```

#### Versions
Linux-4.15.0-74-generic-x86_64-with-debian-buster-sid
Python 3.7.4 (default, Aug 13 2019, 20:35:49) 
[GCC 7.3.0]
NumPy 1.18.1
matplotlib 3.1.3
wordcoud 1.6.0.post27+g260c20f
fontTools 4.3.0
spyder 4.0.0

**Comment by ChristofKaufmann** [comment]
Found the issue. In my opinion, it is an error in fontTools. I will explain what causes the error.

#### Difference between Spyder and pure python

My locale is 'de_DE'. When using Spyder the `LC_TIME` category is set to `'de_DE'`, while in python it is set to `None`. This can be checked with
```python
import locale
print(locale.getlocale(locale.LC_TIME))
# outputs: ('de_DE', 'UTF-8') in Spyder
# outputs: (None, None) in pure python
```
So somehow the Spyder start script changes the locale more consistently than pure python. Note, that starting `jupyter qtconsole` (the console in Spyder) behaves the same as pure python, so it is really Spyder specific.

The locale is used e. g. for the abbreviated week day:
```python
print(locale.nl_langinfo(locale.ABDAY_1))
# outputs: 'So' in Spyder
# outputs: 'Sun' in pure python
```
This changes the behaviour of `time.strptime`:
```python
time.strptime('Mo', '%a')  # works in Spyder, errors in pure python
time.strptime('Mon', '%a') # works in pure python, errors in Spyder
```

#### Following the trace

To embed the subset of the font into the SVG, wordcloud saves the glyphs using fontTools in XML format into a buffer and reads it as WOFF from it again. See [here](https://github.com/amueller/word_cloud/blob/master/wordcloud/wordcloud.py#L866). The reading causes the error.

Going into fontTools now. The XML attributes of the font like 'created' date is parsed in [`fromXML`](https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/ttLib/tables/_h_e_a_d.py#L113). For the 'created' and 'modified' dates it calls [`timestampFromString`](https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/misc/timeTools.py#L47) with `value='Mon Jan  8 12:28:04 2007'` in my case. Inside `timestampFromString` is a `time.strptime(value)`, which tries to parse a localized string, while the date string in the font in not localized. 

#### To do

In my opinion fontTools should parse the date without localization. This is the root cause.

However, you could try to workaround this issue by skipping the intermediate XML step (as you already have as TODO-comment [here](https://github.com/amueller/word_cloud/blob/master/wordcloud/wordcloud.py#L861)). It should be possible to transform directly to WOFF, since fontTools provides a tool specifically for subsetting a font, see here for their [save_font](https://rsms.me/fonttools-docs/_modules/fontTools/subset.html#save_font) function. They don't use XML.

**Comment by ChristofKaufmann** [comment]
I resolved the issue with fonttools/fonttools#1839. I still leave this issue open for now, since you might want to do the workaround. Feel free to close.

**Comment by amueller** [comment]
Thanks for investigating and fixing this upstream. It seems tricky to do a workaround here without rewriting the svg generation, which I don't know if anyone has time to right now. I think it's good to have this issue for future reference but I'll close it for now.

Cheers!
